typescript/listeners/listeners.py

```diff
@@ -227,7 +238,31 @@ def on_close(self, view):
             #         most_recent_used_file_list.remove(info)
             # notify the server that the file is closed
             cli.service.close(file_name)
+        if TypeScriptEventListener.__close_all_is_happening():
+            TypeScriptEventListener.__remove_close_all_view(view)
```

Comment:
This can be simplified. You just need to check if the current number of opened views = 1, which eliminates the need to maintain a list of opened view